src/migration/1759421991433-UserExperienceRevamped.ts

```diff
@@ -0,0 +1,94 @@
+    );
+    await queryRunner.query(
+      `CREATE UNIQUE INDEX "IDX_location_country_subdivision_city_unique" ON "dataset_location" ("country", "subdivision", "city") `,
+    );
+    await queryRunner.query(
+      `CREATE TABLE "user_experience" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "userId" character varying NOT NULL, "companyId" text NOT NULL, "title" text NOT NULL, "subtitle" text, "description" text, "startedAt" TIMESTAMP NOT NULL, "endedAt" TIMESTAMP, "type" text NOT NULL, "locationId" uuid NOT NULL, "locationType" integer array NOT NULL DEFAULT '{1,2,3}', "employmentType" integer array DEFAULT '{1,2,3,4}', "verified" boolean DEFAULT false, "url" text, "grade" text, "externalReferenceId" text, CONSTRAINT "PK_user_experience_id" PRIMARY KEY ("id")); COMMENT ON COLUMN "user_experience"."locationType" IS 'LocationType from protobuf schema'; COMMENT ON COLUMN "user_experience"."employmentType" IS 'EmploymentType from protobuf schema'`,
+    );
+    await queryRunner.query(
+      `CREATE INDEX "IDX_user_experience_userId" ON "user_experience" ("userId") `,
+    );
+    await queryRunner.query(
+      `CREATE INDEX "IDX_user_experience_type" ON "user_experience" ("type") `,
+    );
+    await queryRunner.query(
+      `CREATE TABLE "user_experience_skill" ("slug" text NOT NULL, "experienceId" uuid NOT NULL, CONSTRAINT "PK_COMPOSITE_user_experience_skill_slug_experienceId" PRIMARY KEY ("slug", "experienceId"))`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "user_experience" ADD CONSTRAINT "FK_user_experience_user_userId" FOREIGN KEY ("userId") REFERENCES "user"("id") ON DELETE CASCADE ON UPDATE NO ACTION`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "user_experience" ADD CONSTRAINT "FK_user_experience_company_companyId" FOREIGN KEY ("companyId") REFERENCES "company"("id") ON DELETE CASCADE ON UPDATE NO ACTION`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "user_experience" ADD CONSTRAINT "FK_user_experience_dataset_location_locationId" FOREIGN KEY ("locationId") REFERENCES "dataset_location"("id") ON DELETE SET NULL ON UPDATE NO ACTION`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "user_experience_skill" ADD CONSTRAINT "FK_user_experience_skill_user_skill_slug" FOREIGN KEY ("slug") REFERENCES "user_skill"("slug") ON DELETE CASCADE ON UPDATE NO ACTION`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "user_experience_skill" ADD CONSTRAINT "FK_user_experience_skill_user_experience_experienceId" FOREIGN KEY ("experienceId") REFERENCES "user_experience"("id") ON DELETE CASCADE ON UPDATE NO ACTION`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "company" ADD "type" text NOT NULL DEFAULT 'company'`,
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
```

Comment:
same here, no need to drop constraint before dropping table :)

Reply:
same with indicies

Reply:
This one was purely generated. I'm cool with dropping immediately, just not sure if there are any implications to future generations.